shell/platform/darwin/macos/framework/Source/FLEEngine.mm

```diff
@@ -332,22 +332,25 @@ - (void)shutDownEngine {
 #pragma mark - FlutterBinaryMessenger
 
 - (void)sendOnChannel:(nonnull NSString*)channel message:(nullable NSData*)message {
+  [self sendOnChannel:channel message:message binaryReply:nil];
+}
+
+- (void)sendOnChannel:(NSString*)channel
+              message:(NSData* _Nullable)message
+          binaryReply:(FlutterBinaryReply _Nullable)callback {
   FlutterPlatformMessage platformMessage = {
       .struct_size = sizeof(FlutterPlatformMessage),
       .channel = [channel UTF8String],
       .message = static_cast<const uint8_t*>(message.bytes),
       .message_size = message.length,
   };
-
-  FlutterEngineResult result = FlutterEngineSendPlatformMessage(_engine, &platformMessage);
-  if (result != kSuccess) {
-    NSLog(@"Failed to send message to Flutter engine on channel '%@' (%d).", channel, result);
+  if (!callback) {
```

Comment:
This moves the code duplication into the same method, rather than removing it. We should have a single copy of the code that calls FlutterEngineSendPlatformMessage, checks the return value, and logs. The conditional part should be populating the reply in the message (and then cleaning it up).

Reply:
Done